source/utilities.tex

```diff
@@ -8263,7 +8235,9 @@
 from \tcode{std::forward<decltype(d)>(d)}.
 
 \pnum
-\remarks These constructors shall not participate in overload resolution
+\remarks If \tcode{D} is a reference type,
+the second constructor is defined as deleted.
+These constructors shall not participate in overload resolution
 unless \tcode{is_constructible_v<D, decltype(d)>} is \tcode{true}.
```

Comment:
There's a pre-existing problem here. If `D` is a reference type `A&` then `is_constructible<D, decltype(d)>` is `is_constructible<A&, A&&>` which is false, so it's irrelevant that it's deleted.

Do we want to tweak the `is_constructible` constraint so that the deleted constructor participates in overload resolution for both `A&` and `const A&` cases, so that trying to initialize a deleter of reference type from an rvalue always calls the deleted function?

Reply:
I think we could just change the _Remarks:_ to only apply to non-reference deleters. The constraint never removes the first constructor from overload resolution for a reference deleter, because `is_constructible<A&, A&>` and `is_constructible<const A&, const A&>` are true. For the second constructor we don't want to remove it from overload resolution for references deleters, so that the deleted function gets chosen.

So for both constructors, the _Remarks:_ isn't useful.

Reply:
I'll take that question to LWG. This patch doesn't introduce the problem, and does resolve #1530